### PR TITLE
Update release notes draft to version v1.22.0-rc.0

### PR DESCRIPTION
**Pull request by cici37**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR updates the Release Notes Draft to k/k v1.22.0-rc.0

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
This is an automated PR generated from `krel The Kubernetes Release Toolbox`

### CODE REVIEW

File: releases/release-1.22/release-notes/release-notes-draft.json [hunk 1]
```diff
@@ -3405,8 +3405,8 @@
   },
   "101977": {
     "commit": "2a106464d355d04f1c8931a883d265cf70612a30",
-    "text": "kubeadm: fix the bug that kubeadm only uses the first hash in `caCertHashes` to verify the root CA",
-    "markdown": "Kubeadm: fix the bug that kubeadm only uses the first hash in `caCertHashes` to verify the root CA ([#101977](https://github.com/kubernetes/kubernetes/pull/101977), [@SataQiu](https://github.com/SataQiu))",
+    "text": "kubeadm: fix the bug that `kubeadm` only uses the first hash in `caCertHashes` to verify the root CA.",
+    "markdown": "Kubeadm: fix the bug that `kubeadm` only uses the first hash in `caCertHashes` to verify the root CA. ([#101977](https://github.com/kubernetes/kubernetes/pull/101977), [@SataQiu](https://github.com/SataQiu))",
```
Comment (Damans227) [inline releases/release-1.22/release-notes/release-notes-draft.json:3409]: @pmmalinov01 already took care of this one via a seperate PR: https://github.com/kubernetes/sig-release/pull/1637

Comment (cici37) [reply releases/release-1.22/release-notes/release-notes-draft.json:3409]: Yep. The pr you mentioned fixed the yaml file so the draft json file here could catch the right text. Thanks

Comment (Damans227) [reply releases/release-1.22/release-notes/release-notes-draft.json:3409]: Yup, you are right. I gave it a lgtm. Good catch!!

Comment (pmmalinov01) [reply releases/release-1.22/release-notes/release-notes-draft.json:3409]: It is fine, krel flow update the markdown files :)

**Pull request merged**